orderBook.js

```diff
@@ -1,5 +1,52 @@
-function reconcileOrder(existingBook, incomingOrder) {
-  throw new Error('Not Implemented Yet')
+function getUpdatedBook(existingBook, incomingOrder) {
+  let updatedBook = []
+  let correspondingOrders = existingBook.filter((item) => {
+    return item.type === incomingOrder.type
```

Comment:
Remember that a corresponding order for a buy is a sell and vice verse so you wouldn't want to check that the orders have the same type, rather you'd want them to have different types.